sdk/servicebus/Azure.Messaging.ServiceBus/src/Filters/RuleFilterExtensions.cs

```diff
@@ -21,40 +21,40 @@ public static Filter ParseFromXElement(XElement xElement)
             switch (attribute.Value)
             {
                 case "SqlFilter":
-                    return SqlFilterExtensions.ParseFromXElement(xElement);
+                    return SqlRuleFilterExtensions.ParseFromXElement(xElement);
                 case "CorrelationFilter":
-                    return CorrelationFilterExtensions.ParseFromXElement(xElement);
+                    return CorrelationRuleFilterExtensions.ParseFromXElement(xElement);
                 case "TrueFilter":
-                    return new TrueFilter();
+                    return new TrueRuleFilter();
                 case "FalseFilter":
-                    return new FalseFilter();
+                    return new FalseRuleFilter();
                 default:
                     ServiceBusEventSource.Log.ManagementSerializationException(
-                        $"{nameof(FilterExtensions)}_{nameof(ParseFromXElement)}",
+                        $"{nameof(RuleFilterExtensions)}_{nameof(ParseFromXElement)}",
                         xElement.ToString());
                     return null;
             }
         }
 
-        public static XElement Serialize(this Filter filter)
+        public static XElement Serialize(this RuleFilter filter)
         {
             switch (filter)
             {
-                case SqlFilter sqlFilter:
-                    switch (sqlFilter)
+                case SqlRuleFilter sqlRuleFilter:
+                    switch (sqlRuleFilter)
                     {
-                        case TrueFilter _:
-                            return sqlFilter.Serialize(nameof(TrueFilter));
+                        case TrueRuleFilter _:
+                            return sqlRuleFilter.Serialize("TrueFilter");
 
-                        case FalseFilter _:
-                            return sqlFilter.Serialize(nameof(FalseFilter));
+                        case FalseRuleFilter _:
+                            return sqlRuleFilter.Serialize("FalseFilter");
 
                         default:
-                            return sqlFilter.Serialize(nameof(SqlFilter));
+                            return sqlRuleFilter.Serialize("SqlFilter");
                     }
 
-                case CorrelationFilter correlationFilter:
-                    return correlationFilter.Serialize();
+                case CorrelationRuleFilter correlationRuleFilter:
+                    return correlationRuleFilter.Serialize("CorrelationFilter");
```

Comment:
Hmm, why are we adding this?

Reply:
Service takes filter type as `CorrelationFilter`, so in track-1 we were using `nameof(CorrelationFilter)` in Serialize method. Now we renamed `CorrelationFilter` to `CorrelationRuleFilter`,  that's why I'm passing filter type in this method.

Reply:
We have renamed `CorrelationFilter` to `CorrelationRuleFilter` but service expects `CorrelationFilter` as a filter type that's why I'm passing the type here.

Reply:
I didn't realize that. Do we have a similar issue for any of the other renames?

Reply:
Yes, we send the name what service expects and we expose the name what we want.